### DISCUSSION TITLE
Add support for Webpack's `BannerPlugin`

### DISCUSSION THREAD

**Discussion by guilhermesimoes** [webpack-contrib/webpack-bundle-analyzer#601] [Ideas]
Adding the following [`BannerPlugin`] to a build:

```ts
new BannerPlugin({
  banner: 'console.log("hello world");',
  raw: true,
}),
```

causes webpack-bundle-analyzer [not to return an accurate size][issue]. This is because when webpack-bundle-analyzer parses a bundle it requires the first thing in the bundle to be the start of a webpack module. It would be great if the parsing logic could be changed to take this plugin into account.

[`BannerPlugin`]: https://webpack.js.org/plugins/banner-plugin/
[issue]: https://github.com/webpack-contrib/webpack-bundle-analyzer/issues/598

**Comment by jameshoward** [comment]
We are having a similar issue when using the [Sentry Webpack plugin](https://github.com/getsentry/sentry-javascript-bundler-plugins), which injects a script to the top of the bundle using `BannerPlugin`. This seems to break the `parseBundle` function in `parseUtils.js` because, after minification, the bundle ends up starting like this:

```javascript
!function(){ /* Sentry code here */ }(),(()=>{ /* webpack module here */})();
```

When walking the AST in `parseBundle`, this is matched as a single `ExpressionStatement` but it doesn't match any of the conditions for finding the Webpack module.

This in turn means that `node_modules` are not correctly split out from the `src/` in our project when using this plugin.

There's an example at https://github.com/jameshoward/webpack-sentry-demo but an even simpler reproduction is to inject code with a `BannerPlugin`, provided that code won't be overly-stripped-out by Terser:

```javascript
// webpack.config.js
  …
  plugins: [
    new webpack.BannerPlugin({
      banner: '(function() { console.log(window); })();',
      raw: true
    })
  ]
```
